robertjosephderouin/refactortractor
# Refactoring/filter tag functionality

## Is this a fix or a feature?
- Neither (refactoring)

## What is the change?
- Refactored the "Filter by Recipe Tags" functionality of the 'scripts.js' file
- Each function within this section was refactored to remove unnecessary lines
- Added "filterByRecipe()" handler to utilize helpers so function chaining is unnecessary
- Pulled out filtering functionality - 'findTaggedRecipes()' -  and moved to RecipeRepository

## Where should the reviewer start?
- In the "Filter by Recipe Tags" section of 'scripts.js'
- Then verify that RecipeRepository is a good place for 'getRecipesByTag()'

## How should this be tested?
- Open web server and verify that functionality has not changed and no new errors appear in console
